libcxx/include/__atomic/support.h

```diff
@@ -112,6 +114,9 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 
 template <typename _Tp, typename _Base = __cxx_atomic_base_impl<_Tp> >
 struct __cxx_atomic_impl : public _Base {
+  static_assert(!is_const<_Tp>::value && !is_volatile<_Tp>::value,
+                "std::atomic<T> requires that 'T' be a cv-unqualified type");
```

Comment:
I think we should move the `static_assert` into `__check_atomic_mandates`, like #131754.